browser/src/control/Control.Notebookbar.js

```diff
@@ -303,6 +303,9 @@ L.Control.Notebookbar = L.Control.extend({
 		}
 
 		this.builder.build(shortcutsBar, shortcutsBarData);
+
+		//create SaveState object after addition of shortcut bar in UI
+		this.map.saveState = new app.definitions.saveState(this.map);
```

Comment:
what about compact mode? do we want it there too?
If not - do we check if saveState exists before use? so in case it is missing in compact we don't generate errors

Reply:
> what about compact mode? do we want it there too?

Yes we do

> If not - do we check if saveState exists before use?

Still just for a safe check i did add checks for saveState

> so in case it is missing in compact we don't generate errors

I am adding a new initialize for `saveState` in case if app directly opens in compact mode :)